### PR TITLE
feat: add change user password model

### PR DESCRIPTION
**Pull request by BLasan**
## Description
Modal form for changing user password created.

## Related issues and discussion
#657 

## Screenshots, if any
webapp2

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

**Comment by luckyman20** [comment]
@BLasan Please follow commit style guidelines.

File: src/app/shared/change-password-dialog/change-password-dialog.component.html [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<h1 mat-dialog-title>Change Password</h1>
+<div mat-dialog-content>
+  
+  <form [formGroup]="changePasswordForm">
+    <div fxLayout="row wrap" fxLayout.lt-md="column" fxLayoutGap="2%">
+      <mat-form-field fxFlex="96%">
+        <mat-label>Password</mat-label>
+        <input matInput [(ngModel)]="password" formControlName="password" type="password">
+      </mat-form-field>
+      <mat-error *ngIf="changePasswordForm.controls.password.hasError('required')">
```
Comment (luckyman20) [inline src/app/shared/change-password-dialog/change-password-dialog.component.html]: Put `<mat-error>` inside `<mat-form-field>`.

File: src/app/shared/change-password-dialog/change-password-dialog.component.html [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<h1 mat-dialog-title>Change Password</h1>
+<div mat-dialog-content>
```
Comment (luckyman20) [inline src/app/shared/change-password-dialog/change-password-dialog.component.html]:
```suggestion

<div mat-dialog-content>
```

File: src/app/shared/change-password-dialog/change-password-dialog.component.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+   */
+  changePasswordForm: any;
+  password: string;
+  repeat_password: string;
+
+  /**
+   * @param {MatDialogRef} dialogRef Component reference to dialog.
+   * @param {any} data Provides a deleteContext.
+   */
+  constructor(public dialogRef: MatDialogRef<ChangePasswordDialogComponent>,
```
Comment (luckyman20) [inline src/app/shared/change-password-dialog/change-password-dialog.component.ts]: Please format this function properly. Take a look at other components as to how to format these.

File: src/app/users/users.service.ts [hunk 4]
```diff
@@ -48,6 +49,35 @@ export class UsersService {
...
+            });
+        }
+        return controlToCompare && controlToCompare.value !== c.value ? {'notequal': true} : null;
+    };
+  }
+
+  /**
+   * Change User Password
+   * @param userId
+   * @param password
```
Comment (luckyman20) [inline src/app/users/users.service.ts]:
```suggestion
   * @param password
   * @returns {Observable<any>}
```

File: src/app/users/users.service.ts [hunk 5]
```diff
@@ -48,6 +49,35 @@ export class UsersService {
...
+                subscription.unsubscribe();
+            });
+        }
+        return controlToCompare && controlToCompare.value !== c.value ? {'notequal': true} : null;
+    };
+  }
+
+  /**
+   * Change User Password
+   * @param userId
```
Comment (luckyman20) [inline src/app/users/users.service.ts]: Explain the parameters.

File: src/app/shared/change-password-dialog/change-password-dialog.component.html [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  
+  <form [formGroup]="changePasswordForm">
+    <div fxLayout="row wrap" fxLayout.lt-md="column" fxLayoutGap="2%">
+      <mat-form-field fxFlex="96%">
+        <mat-label>Password</mat-label>
+        <input matInput [(ngModel)]="password" formControlName="password" type="password">
```
Comment (luckyman20) [inline src/app/shared/change-password-dialog/change-password-dialog.component.html]: We aren't working on `[(ngModel)]` approach for handling values of form fields. Please take look on other components as how they are implemented.

File: src/app/users/users.service.ts [hunk 7]
```diff
@@ -48,6 +49,36 @@ export class UsersService {
...
+        const controlToCompare = c.root.get(controlNameToCompare);
+        if (controlToCompare) {
+            const subscription: Subscription = controlToCompare.valueChanges.subscribe(() => {
+                c.updateValueAndValidity();
+                subscription.unsubscribe();
+            });
+        }
+        return controlToCompare && controlToCompare.value !== c.value ? {'notequal': true} : null;
+    };
+  }
```
Comment (karantakalkar) [inline src/app/users/users.service.ts:70]: why is this in service? move it to dialog component.ts, also what did you refer to while writing code for this?

File: src/app/users/view-user/view-user.component.ts [hunk 8]
```diff
@@ -58,4 +59,24 @@ export class ViewUserComponent implements OnInit {
...
+   */
+  changeUserPassword() {
+    const changeUserPasswordDialogRef = this.dialog.open(ChangePasswordDialogComponent, {
+      data: { password: `user ${this.userData.id}` }
+    });
+
+    changeUserPasswordDialogRef.afterClosed().subscribe((response: any) => {
+      if (response.password && response.repeatPassword) {
+        const password = response.password;
+        const repeat_password = response.repeatPassword;
```
Comment (karantakalkar) [inline src/app/users/view-user/view-user.component.ts:74]: use camel casing

File: src/app/users/view-user/view-user.component.ts [hunk 9]
```diff
@@ -58,4 +59,24 @@ export class ViewUserComponent implements OnInit {
...
 
+
+  /**
+   * Change Password of the Users
+   */
+  changeUserPassword() {
+    const changeUserPasswordDialogRef = this.dialog.open(ChangePasswordDialogComponent, {
+      data: { password: `user ${this.userData.id}` }
+    });
+
```
Comment (karantakalkar) [inline src/app/users/view-user/view-user.component.ts:70]:
```suggestion
```

File: src/app/users/view-user/view-user.component.ts [hunk 10]
```diff
@@ -58,4 +59,24 @@ export class ViewUserComponent implements OnInit {
     });
   }
 
+
+  /**
+   * Change Password of the Users
+   */
+  changeUserPassword() {
+    const changeUserPasswordDialogRef = this.dialog.open(ChangePasswordDialogComponent, {
+      data: { password: `user ${this.userData.id}` }
```
Comment (karantakalkar) [inline src/app/users/view-user/view-user.component.ts:68]: can you explain this line.

File: src/app/users/view-user/view-user.component.ts (hunk 9 above)
Comment (karantakalkar) [reply src/app/users/view-user/view-user.component.ts:70]: add comments wherever necessary

File: src/app/shared/shared.module.ts [hunk 11]
```diff
@@ -15,6 +15,7 @@ import { ThemePickerComponent } from './theme-picker/theme-picker.component';
 /** Custom Modules */
 import { IconsModule } from './icons.module';
 import { MaterialModule } from './material.module';
+import { ChangePasswordDialogComponent } from './change-password-dialog/change-password-dialog.component';
```
Comment (karantakalkar) [inline src/app/shared/shared.module.ts:18]: this is not a module import at the right place

File: src/app/shared/change-password-dialog/change-password-dialog.component.ts [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+  ngOnInit() {
+    this.createChangePasswordForm();
+  }
+
+  /**
+   * Create Change Password Form
+   */
+  createChangePasswordForm() {
+    this.changePasswordForm = this.formBuilder.group({
+      'password': ['', [Validators.required, Validators.pattern('(.*[a-z].*)'), Validators.pattern('(.*[A-Z].*)'), Validators.pattern('(.*\d.*)')]],
```
[the remark this answers is not in the document]
Comment (BLasan) [reply src/app/shared/change-password-dialog/change-password-dialog.component.ts:36]: It says that there should be a uppercase and lowercase letters and a numeric character. Isn't it needed? May I remove it?

[the remark this answers is not in the document]
Comment (karantakalkar) [reply src/app/shared/change-password-dialog/change-password-dialog.component.ts:36]: change for now we can always make it strict later

File: src/app/users/users.service.ts (hunk 7 above)
Comment (BLasan) [reply src/app/users/users.service.ts:70]: Tried to compare passwor and re-enter password features. Can I move this to component.ts file?

File: src/app/shared/change-password-dialog/change-password-dialog.component.ts (hunk 12 above)
[the remark this answers is not in the document]
Comment (BLasan) [reply src/app/shared/change-password-dialog/change-password-dialog.component.ts:36]: Okay. I removed the validators for the pattern checking

[the remark this answers is not in the document]
Comment (BLasan) [reply src/app/shared/change-password-dialog/change-password-dialog.component.ts:36]: Screenshot from 2020-05-13 18-02-09

I think that validations are needed

[the remark this answers is not in the document]
Comment (karantakalkar) [reply src/app/shared/change-password-dialog/change-password-dialog.component.ts:36]: what is the pattern rule?

File: src/app/users/view-user/view-user.component.ts (hunk 10 above)
Comment (BLasan) [reply src/app/users/view-user/view-user.component.ts:68]: passing data to the mat-dialog. But here it's not needed. I have removed that. I'm currently fixing the regex pattern

File: src/app/shared/change-password-dialog/change-password-dialog.component.ts (hunk 12 above)
[the remark this answers is not in the document]
Comment (BLasan) [reply src/app/shared/change-password-dialog/change-password-dialog.component.ts:36]: one uppercase,lower case with a number and the number of character should be at least 8 and no more than 50

File: src/app/shared/change-password-dialog/change-password-dialog.component.html [hunk 13]
```diff
@@ -0,0 +1,33 @@
+<h1 mat-dialog-title>Change Password</h1>
+<div mat-dialog-content>
+  <form [formGroup]="changePasswordForm">
+    <div fxLayout="row wrap" fxLayout.lt-md="column" fxLayoutGap="2%">
```
[the remark this answers is not in the document]
Comment (BLasan) [reply src/app/shared/change-password-dialog/change-password-dialog.component.html:7]: Done. Please have a look

File: src/app/shared/change-password-dialog/change-password-dialog.component.ts (hunk 12 above)
[the remark this answers is not in the document]
Comment (BLasan) [reply src/app/shared/change-password-dialog/change-password-dialog.component.ts]: This is fixed. Please check

File: src/app/users/users.service.ts (hunk 7 above)
Comment (BLasan) [reply src/app/users/users.service.ts]: Added to the component.ts file

File: src/app/shared/change-password-dialog/change-password-dialog.component.ts [hunk 14]
```diff
@@ -0,0 +1,41 @@
+import { Component, OnInit, Inject } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material';
+import { UsersService } from 'app/users/users.service';
+import { FormBuilder, Validators } from '@angular/forms';
+@Component({
```
[the remark this answers is not in the document]
Comment (BLasan) [reply src/app/shared/change-password-dialog/change-password-dialog.component.ts:10]: This is done. Please have a look

File: src/app/shared/change-password-dialog/change-password-dialog.component.ts [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+  selector: 'mifosx-change-password-dialog',
+  templateUrl: './change-password-dialog.component.html',
+  styleUrls: ['./change-password-dialog.component.scss']
+})
+export class ChangePasswordDialogComponent implements OnInit {
+
+  /**
+   * Change Password Form
+   */
```
[the remark this answers is not in the document]
Comment (BLasan) [reply src/app/shared/change-password-dialog/change-password-dialog.component.ts:19]: Please review

**Comment by karantakalkar** [comment]
@BLasan please resolve all conversations after fixing

File: src/app/users/users.service.ts (hunk 7 above)
Comment (karantakalkar) [reply src/app/users/users.service.ts]: sorry I forgot to delete this comment we will be having a password service containing these 2 functions

File: src/app/shared/change-password-dialog/change-password-dialog.component.ts (hunk 15 above)
[the remark this answers is not in the document]
Comment (karantakalkar) [reply src/app/shared/change-password-dialog/change-password-dialog.component.ts:19]: its still in 3 line

File: src/app/shared/change-password-dialog/change-password-dialog.component.ts (hunk 12 above)
[the remark this answers is not in the document]
Comment (karantakalkar) [reply src/app/shared/change-password-dialog/change-password-dialog.component.ts]: add relevant message in mat error, how will the user know?

File: src/app/users/users.service.ts (hunk 7 above)
Comment (karantakalkar) [reply src/app/users/users.service.ts]: keep password service in same folder as this dialog and keep the dialog in shared folder

Comment (BLasan) [reply src/app/users/users.service.ts]: Added a new service file. Please review

**Comment by karantakalkar** [comment]
@BLasan lete me test it on my machine till then can you synchronize self service user password.

**Comment by BLasan** [comment]
> @BLasan let me test it on my machine till then can you synchronize self service user password.

Sure, I'm working on that

File: src/app/shared/change-password-dialog/change-password-dialog.component.ts [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+
+  ngOnInit() {
+    this.createChangePasswordForm();
+  }
+
+  /** Change Password form */
+  createChangePasswordForm() {
+    this.changePasswordForm = this.formBuilder.group({
+      'password': ['', [Validators.required, Validators.pattern('^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9]).{8,}$'), Validators.maxLength(50), Validators.minLength(8)]],
+      'repeat_password': ['', [Validators.required, this.passwordCheckService.confirmPassword('password')]]
```
Comment (karantakalkar) [inline src/app/shared/change-password-dialog/change-password-dialog.component.ts:41]: repeatPassword update everywhere

File: src/app/shared/change-password-dialog/password-check.service.ts [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+        const controlToCompare = c.root.get(controlNameToCompare);
+        if (controlToCompare) {
+            const subscription: Subscription = controlToCompare.valueChanges.subscribe(() => {
+                c.updateValueAndValidity();
+                subscription.unsubscribe();
+            });
+        }
+        return controlToCompare && controlToCompare.value !== c.value ? {'notequal': true} : null;
+    };
+  }
```
Comment (karantakalkar) [inline src/app/shared/change-password-dialog/password-check.service.ts:34]: can you tell me what did you refer to while making this function

File: src/app/users/users.service.ts [hunk 18]
```diff
@@ -5,6 +5,7 @@ import { HttpClient, HttpParams } from '@angular/common/http';
 /** rxjs Imports */
 import { Observable } from 'rxjs';
 
+
```
Comment (karantakalkar) [inline src/app/users/users.service.ts:8]:
```suggestion
```

File: src/app/shared/change-password-dialog/change-password-dialog.component.ts [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+@Component({
+  selector: 'mifosx-change-password-dialog',
+  templateUrl: './change-password-dialog.component.html',
+  styleUrls: ['./change-password-dialog.component.scss']
+})
+export class ChangePasswordDialogComponent implements OnInit {
+
+  /**
+   * Change Password Form
+   */
```
Comment (karantakalkar) [inline src/app/shared/change-password-dialog/change-password-dialog.component.ts:21]:
```suggestion
/** Change Password Form */
```

File: src/app/shared/change-password-dialog/change-password-dialog.component.ts [hunk 20]
```diff
@@ -0,0 +1,45 @@
+/** Angular Imports */
+import { Component, OnInit, Inject } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material';
+import { FormBuilder, Validators } from '@angular/forms';
+
+/** Custom Imports */
+import { PasswordCheckService } from './password-check.service';
+
```
Comment (karantakalkar) [inline src/app/shared/change-password-dialog/change-password-dialog.component.ts:8]: please change the width and height to offer better ux
Screenshot from 2020-05-13 20-24-05

File: src/app/users/users.service.ts [hunk 21]
```diff
@@ -48,6 +49,16 @@ export class UsersService {
...
+  /**
+   * Change User Password
+   * @param userId User Id of users
+   * @param password New Password of the user
+   * @returns {Observable<any>}
+   */
+  changePassword(userId: string, passwordObj: any) {
+    return this.http.put(`/users/${userId}`, passwordObj);
+  }
+
```
[the remark this answers is not in the document]
Comment (BLasan) [reply src/app/users/users.service.ts:61]: This function too?

File: src/app/shared/change-password-dialog/password-check.service.ts (hunk 17 above)
Comment (BLasan) [reply src/app/shared/change-password-dialog/password-check.service.ts:34]: Referred to a stack overflow solution. I edited this according to my problem

File: src/app/shared/change-password-dialog/change-password-dialog.component.ts (hunk 20 above)
Comment (BLasan) [reply src/app/shared/change-password-dialog/change-password-dialog.component.ts:8]: Can you put the required width and the height?

Comment (BLasan) [reply src/app/shared/change-password-dialog/change-password-dialog.component.ts:8]: Screenshot from 2020-05-13 21-44-56

Is this okay?

Comment (BLasan) [reply src/app/shared/change-password-dialog/change-password-dialog.component.ts:8]: @karantakalkar Please review

Comment (karantakalkar) [reply src/app/shared/change-password-dialog/change-password-dialog.component.ts:8]: good

File: src/app/shared/change-password-dialog/password-check.service.ts (hunk 17 above)
Comment (karantakalkar) [reply src/app/shared/change-password-dialog/password-check.service.ts]: can you find the link please?

File: src/app/shared/change-password-dialog/change-password-dialog.component.html [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+        </mat-error>
+      </mat-form-field>
+
+    </div>
+
+  </form>
+
+</div>
+
+<mat-dialog-actions align="end">
```
Comment (karantakalkar) [inline src/app/shared/change-password-dialog/change-password-dialog.component.html:44]: center align the buttons

File: src/app/shared/change-password-dialog/change-password-dialog.component.html [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+  <button mat-raised-button mat-dialog-close>Cancel</button>
+  <button mat-raised-button color="warn" [disabled]="!changePasswordForm.valid" [mat-dialog-close]="{password:changePasswordForm.get('password').value,repeatPassword:changePasswordForm.get('repeatPassword').value}">Confirm</button>
```
Comment (karantakalkar) [inline src/app/shared/change-password-dialog/change-password-dialog.component.html:46]: get mat dialog close attribute as a return value of a function

File: src/app/shared/change-password-dialog/change-password-dialog.component.html [hunk 24]
```diff
@@ -0,0 +1,47 @@
...
+        <mat-error *ngIf="changePasswordForm.controls.password.hasError('pattern')">
+          Pattern does not match
+        </mat-error>
+      </mat-form-field>
+
+    </div>
+
+    <div fxLayout="row wrap" fxLayout.lt-md="column" fxLayoutGap="2%">
+
+      <mat-form-field fxFlex="96%">
```
Comment (karantakalkar) [inline src/app/shared/change-password-dialog/change-password-dialog.component.html:30]: remove fxFlex=96

File: src/app/shared/change-password-dialog/password-check.service.ts (hunk 17 above)
Comment (BLasan) [reply src/app/shared/change-password-dialog/password-check.service.ts]: https://stackoverflow.com/questions/51605737/confirm-password-validation-in-angular-6

Scroll down. You'll find a similar solution for this. I can't remember the exact link as I found this solution long time ago.

Comment (BLasan) [reply src/app/shared/change-password-dialog/password-check.service.ts]: Is this okay?

File: src/app/shared/change-password-dialog/change-password-dialog.component.html (hunk 23 above)
Comment (BLasan) [reply src/app/shared/change-password-dialog/change-password-dialog.component.html:46]: It's been used

File: src/app/shared/change-password-dialog/change-password-dialog.component.html [hunk 25]
```diff
@@ -0,0 +1,47 @@
...
+          Password <strong>is required</strong>
+        </mat-error>
+        <mat-error *ngIf="changePasswordForm.controls.password.hasError('minlength')">
+          Password should be at least <strong>8 characters long</strong>
+        </mat-error>
+        <mat-error *ngIf="changePasswordForm.controls.password.hasError('maxlength')">
+          Password should not be more than <strong>50 characters long</strong>
+        </mat-error>
+        <mat-error *ngIf="changePasswordForm.controls.password.hasError('pattern')">
+          Pattern does not match
```
Comment (karantakalkar) [inline src/app/shared/change-password-dialog/change-password-dialog.component.html:22]: change error to "password should include a **numeral** and at **least one uppercase, lowercase and special character**"

File: src/app/shared/change-password-dialog/password-check.service.ts (hunk 17 above)
Comment (BLasan) [reply src/app/shared/change-password-dialog/password-check.service.ts]: Ok

File: src/app/shared/change-password-dialog/change-password-dialog.component.html (hunk 25 above)
Comment (BLasan) [reply src/app/shared/change-password-dialog/change-password-dialog.component.html:22]: Can I give the maximum and minimum number of characters with in the pattern by replacing Validators.maxLength and Validators.minLength ?

Comment (karantakalkar) [reply src/app/shared/change-password-dialog/change-password-dialog.component.html:22]: no its fine

**Comment by karantakalkar** [comment]
@muskankhedia please review

File: src/app/shared/change-password-dialog/password.service.ts [hunk 26]
```diff
@@ -0,0 +1,50 @@
...
+        const controlToCompare = c.root.get(controlNameToCompare);
+        if (controlToCompare) {
+            const subscription: Subscription = controlToCompare.valueChanges.subscribe(() => {
+                c.updateValueAndValidity();
+                subscription.unsubscribe();
+            });
+        }
+        return controlToCompare && controlToCompare.value !== c.value ? {'notequal': true} : null;
+    };
+  }
```
Comment (muskankhedia) [inline src/app/shared/change-password-dialog/password.service.ts:38]: The function can be executed in the `change-password-dialog.component.ts`, a separate file is not needed. Can you change this?

File: src/app/shared/change-password-dialog/password.service.ts [hunk 27]
```diff
@@ -0,0 +1,50 @@
...
+
+  /**
+   * Change User Password
+   * @param userId User Id of users
+   * @param password New Password of the user
+   * @returns {Observable<any>}
+   */
+  changePassword(userId: string, passwordObj: any) {
+    return this.http.put(`/users/${userId}`, passwordObj);
+  }
```
[the remark this answers is not in the document]
Comment (karantakalkar) [reply src/app/shared/change-password-dialog/password.service.ts:48]: It is being used in self service users as well see #670

File: src/app/shared/change-password-dialog/password.service.ts (hunk 26 above)
Comment (karantakalkar) [reply src/app/shared/change-password-dialog/password.service.ts:38]: This is being used in self service user password as well

Comment (karantakalkar) [reply src/app/shared/change-password-dialog/password.service.ts:38]: Muskan please dounle check validatior function thats the main thing

File: src/app/shared/change-password-dialog/change-password-dialog.component.ts [hunk 28]
```diff
@@ -0,0 +1,43 @@
...
+  ngOnInit() {
+    this.createChangePasswordForm();
+  }
+
+  /** Change Password form */
+  createChangePasswordForm() {
+    this.changePasswordForm = this.formBuilder.group({
+      'password': ['', [Validators.required, Validators.pattern('^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*[$@$!%*?&])[A-Za-z\d$@$!%*?&].{8,}$'), Validators.maxLength(50), Validators.minLength(8)]],
+      'repeatPassword': ['', [Validators.required, this.passwordCheckService.confirmPassword('password')]]
```
Comment (karantakalkar) [inline src/app/shared/change-password-dialog/change-password-dialog.component.ts:39]: password service

Comment (karantakalkar) [reply src/app/shared/change-password-dialog/change-password-dialog.component.ts:39]: @BLasan Muskan will guide you through this

Comment (BLasan) [reply src/app/shared/change-password-dialog/change-password-dialog.component.ts:39]: Ok. Any errors encountered?

File: src/app/shared/change-password-dialog/password.service.ts (hunk 27 above)
[the remark this answers is not in the document]
Comment (BLasan) [reply src/app/shared/change-password-dialog/password.service.ts:48]: I'll fix this

File: src/app/shared/change-password-dialog/password.service.ts (hunk 26 above)
Comment (BLasan) [reply src/app/shared/change-password-dialog/password.service.ts:38]: Yuh. May I delete the service file ? But this will be needed in #670 too

Comment (karantakalkar) [reply src/app/shared/change-password-dialog/password.service.ts:38]: put this in dialog we'll use the dialog in 670 too

File: src/app/shared/change-password-dialog/change-password-dialog.component.ts (hunk 28 above)
Comment (karantakalkar) [reply src/app/shared/change-password-dialog/change-password-dialog.component.ts:39]: it's not passwordcheckservice it's password service

Comment (BLasan) [reply src/app/shared/change-password-dialog/change-password-dialog.component.ts]: I moved the file to component.ts . Now there's no service file

**Comment by BLasan** [comment]
@muskankhedia  Please review this

File: src/app/users/view-user/view-user.component.ts [hunk 29]
```diff
@@ -58,4 +59,23 @@ export class ViewUserComponent implements OnInit {
...
+  changeUserPassword() {
+    const changeUserPasswordDialogRef = this.dialog.open(ChangePasswordDialogComponent, {
+      width: '400px',
+      height: '300px'
+    });
+    changeUserPasswordDialogRef.afterClosed().subscribe((response: any) => {
+      if (response.password && response.repeatPassword) {
+        const password = response.password;
+        const repeatPassword = response.repeatPassword;
+        this.usersService.changePassword(this.userData.id, {password: password, repeatPassword: repeatPassword}).subscribe(() => {
```
Comment (muskankhedia) [inline src/app/users/view-user/view-user.component.ts:74]: Please follow the documentation to send the data to the API [Link](https://demo.mifos.io/api-docs/apiLive.htm#users_update).

You need to add another parameter, `firstname` in the data to make a successful request. And also make another object named `data` with all the fields and you can directly send that to the API.

Comment (BLasan) [reply src/app/users/view-user/view-user.component.ts]: Done :) Please review

**Pull request merged**